Add additional Youtube info

spaantje opened the pull request:
Hi @christophrumpel, 

This PR add some additional info to the Streams. 

### The "Real" start time of the stream (#52). 
I do nothing with this data at the moment, but we could use this on the archive page instead of the planned time? 
I also added the Ended time so you could display the length of the stream. 

### Removed deleted/private streams (#50). 
I also "delete" any streams for which we cant get any info. Well actually i don't delete them, but set the status to 'deleted' so then don't show up on the site. (In case we delete any which should not be deleted. 
Also i added the `hidden_at` timestamp to the database so we know when it was "deleted".

I created a command which runs once a day to update the latest 50 archived streams. 
I limit it to 50 to account for the limit on the Youtube API. Maybe we should process more, but i don't know if that is necessary?

Fixes #50, Fixes #52

christophrumpel commented:
Hey @spaantje, that's great. Showing durations would be perfect too. That way we can also show statistics later, like who streamed how longs this week, months, etc. Can you check the conflicts? If you need help resolving them, let me know. Then I  take a closer look at the PR. 👍

spaantje commented:
Hi @christophrumpel, I have fixed the conflict and added the duration to the archive:
![image](https://user-images.githubusercontent.com/477362/125960804-0c3be4e8-6546-43eb-8322-0dada0050e95.png)

Let me know if the design is ok.

christophrumpel commented:
Thanks. I just took a look and I realized I need to adapt the current commands, it is not clear anymore what they do (not related to your code).

Here is how I would change them:


**Command: "UpdateUpcomingStreams"**
Goal: get new titles, descriptions, etc.
Query: status upcoming
Update: whole stream
Run: every hour

**Command "UpdateStreamsThatGoLive"**
Goal: set them to live
Query: status upcoming + planned start in <= 15min
Update: set to live if they are
Run: every minute

**Command "UpdateEndedStreams"** 
Goal: set them to past and get ended stream info
Query: streams with status live
Update: set to ended + update ended info (start + end time etc.)
Run: every 5 minutes


Here are our options: 
* I change the commands in the main branch without the additions of the newly collected data you suggested, you would then only need to fix the conflicts in this branch
* I do those changes directly here in your PR and you then finish it
* You add those changes to this PR (which would be quite some work because we need to adapt some tests too)

So first, what do you think about the suggested changes? And 2nd how do you want to move on here. I'm fine with all options, just let me know what works best for you.

spaantje commented:
@christophrumpel I don't mind if you update them or if I do them. 
I think it's best to do the changes in this PR however (it would be a lot of merge conflict otherwise). 

**About the changes:**

I would add another command (or flag). 

**Command "UpdateArchivedStreams"**
Goal: delete missing streams and update any changes to title/description
Query: last 50 streams with status finished
Update: status to deleted if deleted; whole stream
Run: once a day

Maybe change some names: 

UpdateStreamsThatGoLive --> UpdateAlmostLiveStreams
UpdateEndedStreams --> UpdateNowLiveStreams
(And suffix all or non with `Command` according to preference.)
That way all of them follow the same naming "Update<query>Stream" 

For the UpdateStreamsThatGoLive / UpdateAlmostLiveStreams maybe limit the query to 10 or maybe even 5 minutes. How many time will someone really start before the time? And it has a high impact on your API key otherwise. 

And maybe we should move the updating of the Stream itself to another place (to the Stream model or to a Action)? We would end up with a lot of duplication otherwise.

I don't know i'm able to finish the PR today (before your stream). So if you would like to merge today maybe it's best if you do the changes? Otherwise i will be able to check tomorrow.

christophrumpel commented on app/Console/Commands/UpdateArchivedStreamsCommand.php:
```diff
@@ -0,0 +1,69 @@
+{
+    protected $signature = 'larastreamers:update-archived-streams';
+
+    protected $description = 'Update latest 50 archived (& live) streams';
```
I like the idea to only check the currently live streams in the dedicated command, so in this command, we can focus on the finished streams. This way we can check finished streams at the end of every day.

PS: Currently finished streams will directly go to the archive, but after this PR I change this so they stay on the home page till the next day and we mark them as finished. This way we don't have to update the finished streams right away.

spaantje replied:
But i do still think we need to include the streams with the `live` status because of issue #50. (Sometimes a stream get's set to private before the `CheckIfLiveStreamsHaveEndedCommand` has run and then they don't get updated.)

christophrumpel replied:
Ok, good point. Then let us add the live streams to that command as well, since we don't have a lot of current leave streams. Can you add that?

spaantje replied:
Yes, is included.

christophrumpel commented:
Thanks a lot on this PR @spaantje and your patience :-)
